evap/rewards/tests/test_views.py

```diff
@@ -57,16 +57,16 @@ def test_redeem_all_points(self):
         form.set(f"points-{self.event1.pk}", 2)
         form.set(f"points-{self.event2.pk}", 3)
         response = form.submit()
-        self.assertContains(response, "You successfully redeemed your points.")
+        self.assertContains(response, "You successfully redeemed your points.", status_code=200)
         self.assertEqual(0, reward_points_of_user(self.student))
 
     def test_redeem_too_many_points(self):
         response = self.app.get(self.url, user=self.student)
         form = response.forms["reward-redemption-form"]
         form.set(f"points-{self.event1.pk}", 3)
         form.set(f"points-{self.event2.pk}", 3)
-        response = form.submit()
-        self.assertContains(response, "have enough reward points.")
+        response = form.submit(expect_errors=True)
+        self.assertContains(response, "have enough reward points.", status_code=400)
```

Comment:
Have you tried this?
```suggestion
        response = form.submit(status=400)
        self.assertContains(response, "have enough reward points.")
```

I think this would be the most canonical code here.

(repeated below)

Reply:
While the parameter `status=400` stops the failure on submit, it still has to be set in `assertContains`, as it also expects a status code of 200.

Reply:
Fair enough, the code is fine then :)